pkg/action/cl001/ac010/executor.go

```diff
@@ -2,8 +2,86 @@ package ac010
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	batchapiv1 "k8s.io/api/batch/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/awscnfm/v12/pkg/client"
+	"github.com/giantswarm/awscnfm/v12/pkg/env"
+	"github.com/giantswarm/awscnfm/v12/pkg/project"
+)
+
+const (
+	kubeSystemNamespace = "kube-system"
 )
 
 func (e *Executor) execute(ctx context.Context) error {
+	var err error
+
+	var cpClients k8sclient.Interface
+	{
+		c := client.ControlPlaneConfig{
+			Logger: e.logger,
+
+			KubeConfig: env.KubeConfig(),
+		}
+
+		cpClients, err = client.NewControlPlane(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var tcClients k8sclient.Interface
+	{
+		c := client.TenantClusterConfig{
+			ControlPlane: cpClients,
+			Logger:       e.logger,
+
+			Scope: e.scope,
+		}
+
+		tcClients, err = client.NewTenantCluster(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	err = e.checkKiamAWScallPod(ctx, tcClients.K8sClient())
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// checkKiamAWScallPod will check if the AWS API call job was successful
+func (e *Executor) checkKiamAWScallPod(ctx context.Context, tcClient kubernetes.Interface) error {
+
+	name := fmt.Sprintf("%s-kiam-test", project.Name())
+
+	job, err := tcClient.BatchV1().Jobs(kubeSystemNamespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if !isJobCompleted(job) {
+		return microerror.Mask(jobNotCompleted)
+	}
+
 	return nil
 }
+
+func isJobCompleted(j *batchapiv1.Job) bool {
+	for _, c := range j.Status.Conditions {
+		if c.Type == "Complete" && c.Status == apiv1.ConditionTrue {
```

Comment:
How do we know that the AWS calling was successful? Because this here just says the job status condition is `Complete`, which might be true for a lot of different reasons. Does the job ensure it fails somehow in case it cannot connect to the AWS API? In https://github.com/giantswarm/awscnfm/pull/171 I see we execute some command though I am not sure about the exit code propagation and what that then means for Kubernetes.

Reply:
the command from aws-cli  will exit with `0` if it was successful, if it fails it will have non-zero exit code and that is propagated to k8s and as status `Error`, Job is only `Completed` if the container exits with `0`

it's easily tested by running job like this
```
apiVersion: batch/v1
kind: Job
metadata:
  name: test-aws
spec:
  template:
    spec:
      containers:
      - name: test-aws
        image: quay.io/giantswarm/awscli:2.0.24
        command: ["/bin/sh", "-c", "/usr/local/bin/aws route53 list-hosted-zones"]
      restartPolicy: Never
  backoffLimit: 4

```

since it does not have a proper kiam role it will fail and the result on the job will be h
```
  status:
    conditions:
    - lastProbeTime: "2020-09-30T15:10:46Z"
      lastTransitionTime: "2020-09-30T15:10:46Z"
      message: Job has reached the specified backoff limit
      reason: BackoffLimitExceeded
      status: "True"
      type: Failed
    failed: 5
    startTime: "2020-09-30T15:02:33Z"

```